pallets/rmrk-core/src/functions.rs

```diff
@@ -233,6 +233,9 @@ where
 	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
 		ensure!(max_recursions > 0, Error::<T>::TooManyRecursions);
 		Nfts::<T>::remove(collection_id, nft_id);
+		for resource in Resources::<T>::iter_prefix((collection_id, nft_id)) {
+			Resources::<T>::remove((collection_id, nft_id, resource.1.id));
+		}
```

Comment:
Mutating the storage map while iterating over it. Not sure if this is problematic. I asked a question in substarte stackexchange:

https://substrate.stackexchange.com/questions/573/is-it-safe-to-delete-elements-when-iterating-through-a-storage-map-in-substrate

Reply:
According to the answer, we should use `drain_prefix`

Reply:
Good to know, also found this usage pattern with `drain_prefix` in uniques pallet

Reply:
Oh, while using `drain_prefix` we don't need to call `Resource::<T>::remove` any more

Reply:
Got it, something like that then?
```
for _ in Resources::<T>::drain_prefix((collection_id, nft_id)) {}		
```

Reply:
We can safely ignore its return value like this:

```
Resources::<T>::drain_prefix((collection_id, nft_id));
```